dbms/src/TestUtils/ExecutorTestUtils.cpp

```diff
@@ -145,41 +175,58 @@ void ExecutorTest::executeExecutor(
         {
             context.context.setSetting("max_block_size", Field(static_cast<UInt64>(block_size)));
             auto res = executeStreams(request, concurrency);
-            auto test_info_msg = [&]() {
-                const auto & test_info = testing::UnitTest::GetInstance()->current_test_info();
-                assert(test_info);
-                return fmt::format(
-                    "test info:\n"
-                    "    file: {}\n"
-                    "    line: {}\n"
-                    "    test_case_name: {}\n"
-                    "    test_func_name: {}\n"
-                    "    enable_planner: {}\n"
-                    "    concurrency: {}\n"
-                    "    block_size: {}\n"
-                    "    dag_request: \n{}"
-                    "    result_block: \n{}",
-                    test_info->file(),
-                    test_info->line(),
-                    test_info->test_case_name(),
-                    test_info->name(),
-                    enable_planner,
-                    concurrency,
-                    block_size,
-                    ExecutorSerializer().serialize(request.get()),
-                    getColumnsContent(res));
-            };
-            ASSERT_TRUE(assert_func(res)) << test_info_msg();
+            ASSERT_TRUE(assert_func(res)) << testInfoMsg(request, enable_planner, enable_pipeline, concurrency, block_size);
         }
     }
     WRAP_FOR_TEST_END
 }
 
+void ExecutorTest::checkBlockSorted(
+    const std::shared_ptr<tipb::DAGRequest> & request,
+    const SortDescription & sort_desc,
+    std::function<::testing::AssertionResult(const ColumnsWithTypeAndName &, const ColumnsWithTypeAndName &)> assert_func)
+{
+    WRAP_FOR_TEST_BEGIN
+    std::vector<size_t> concurrencies{2, 5, 10};
+    for (auto concurrency : concurrencies)
+    {
+        auto expected_res = executeStreams(request, concurrency);
+        std::vector<size_t> block_sizes{1, 2, DEFAULT_BLOCK_SIZE};
+        for (auto block_size : block_sizes)
+        {
+            context.context.setSetting("max_block_size", Field(static_cast<UInt64>(block_size)));
+            auto return_blocks = getExecuteStreamsReturnBlocks(request, concurrency);
+
+            if (enable_planner)
+            {
+                for (auto & block : return_blocks)
+                    ASSERT_TRUE(isAlreadySorted(block, sort_desc)) << testInfoMsg(request, enable_planner, enable_pipeline, concurrency, block_size);
+            }
+
+            auto res = mergeBlocks(std::move(return_blocks)).getColumnsWithTypeAndName();
+            ASSERT_TRUE(assert_func(expected_res, res)) << testInfoMsg(request, enable_planner, enable_pipeline, concurrency, block_size);
```

Comment:
```suggestion
            for (auto & block : return_blocks)
                    ASSERT_TRUE(isAlreadySorted(block, sort_desc)) << testInfoMsg(request, enable_planner, enable_pipeline, concurrency, block_size);
```
It seems that there is no need to distinguish whether pipeline